src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -624,6 +624,40 @@ def b64_to_hex(s):
     return hex_data
 
 
+def todict(obj, post_processor=None):
+    """
+    Convert an object to a dictionary. Use 'post_processor(original_obj, dictionary)' to update the
+    dictionary in the process
+    """
+    from datetime import date, time, datetime, timedelta
+    from enum import Enum
+    if isinstance(obj, dict):
+        result = {k: todict(v, post_processor) for (k, v) in obj.items()}
+        return post_processor(obj, result) if post_processor else result
+    if isinstance(obj, list):
+        return [todict(a, post_processor) for a in obj]
+    if isinstance(obj, Enum):
+        return obj.value
+    if isinstance(obj, (date, time, datetime)):
+        return obj.isoformat()
+    if isinstance(obj, timedelta):
+        return str(obj)
+    # This is the only difference with knack.util.todict because for typespec generated SDKs
+    # The base model stores data in obj.__dict__['_data'] instead of in obj.__dict__
+    # We need to call obj.as_dict() to extract data for this kind of model
+    if hasattr(obj, 'as_dict') and not hasattr(obj, '_attribute_map'):
```

Comment:
Another approach to detect if `obj` is an SDK model is something similar to

```py
type(obj).__base__.__name__ == 'Model'
```

`str(type(obj).__base__)` returns

```py
"<class 'azure.cli.command_modules.keyvault.vendored_sdks.azure_keyvault_securitydomain._model_base.Model'>"
```

However, this is also not very reliable as the name `Model` may change at any time.

Using `isinstance()` is impossible as `Model` is defined in each SDK, such as `azure.cli.command_modules.keyvault.vendored_sdks.azure_keyvault_securitydomain._model_base.Model`, instead of in `azure.core`. There is no single `Model` that is the base class of all SDK objects.

Also, using `isinstance()` requires importing the `Model` class which impairs the performance.

As an example, `remove_additional_prop_layer` previously imports `Model` with

```py
from msrest.serialization import Model
```

https://github.com/Azure/azure-cli/pull/13843 changed it to use a `try... catch...` strategy:

https://github.com/Azure/azure-cli/blob/375b34d2c5016f3bff79f56134e070a12f9dfb05/src/azure-cli-core/azure/cli/core/commands/__init__.py#L899-L908

Thus, there is no need to import `Model` for non-`Model` objects.